src/os/bluestore/BlueStore.cc

```diff
@@ -4460,7 +4461,9 @@ int BlueStore::fsck(bool deep)
       }
       for (auto& s : o->extent_map.shards) {
 	dout(20) << __func__ << "    shard " << *s.shard_info << dendl;
-	expecting_shards.push_back(s.key);
+	string key;
+	get_extent_shard_key(o->key, s.offset, &key);
+	expecting_shards.push_back(key);
```

Comment:
Yuck. How about:

expecting_shards.resize(expecting_shards.size()+1);
get_extent_shard_key(o->key,s.offset,&expecting_shards[expecting_shards.size()-1]);

This avoids the extra malloc/dealloc for the string on the stack.